hw/ip/padctrl/fpv/padctrl_fpv.core

```diff
@@ -12,22 +12,31 @@ filesets:
       # from a particular top-level config
       - lowrisc:ip:padctrl
       - lowrisc:prim:prim_pkg
+      - lowrisc:fpv:csr_assert_gen
     files:
       - tb/padctrl_fpv.sv
       - tb/padctrl_bind_fpv.sv
       - vip/padctrl_assert_fpv.sv
       - vip/padring_assert_fpv.sv
-      - vip/padctrl_csr_assert_fpv.sv
     file_type: systemVerilogSource
 
+generate:
+  csr_assert_gen:
+    generator: csr_assert_gen
+    parameters:
+      spec: ../data/padctrl.hjson
+      depend: lowrisc:ip:padctrl
```

Comment:
We should be running FPV using IP (generic) as well as top_earlgrey specific libraries. You may need to find a way to use `lowrisc:top_earlgrey:padctrl_reg` for the top level. We can discuss in more detail.

Reply:
Oh thank you for the reminder and help, I will follow the similar way as how rv_plic is done. If I encountered any issues, I will seek your hep further! Thanks Sri.

Reply:
@sriyerg thank you for providing the top level suggestions, do you mind if I implement the top level padctrl and pinmux in the separate PR after Tim's PR #2992 merged?

Reply:
Yeah, sure. Works for me.